pkg/gitreceive/controller.go

```diff
@@ -0,0 +1,150 @@
+package gitreceive
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/deis/builder/pkg"
+	"github.com/deis/builder/pkg/log"
+)
+
+var (
+	potentialExploit = regexp.MustCompile(`\(\)\s+\{[^\}]+\};\s+(.*)`)
+)
+
+type unexpectedControllerStatusCode struct {
+	endpoint string
+	expected int
+	actual   int
+}
+
+func newUnexpectedControllerStatusCode(endpoint string, expectedCode, actualCode int) unexpectedControllerStatusCode {
+	return unexpectedControllerStatusCode{endpoint: endpoint, expected: expectedCode, actual: actualCode}
+}
+
+func (u unexpectedControllerStatusCode) Error() string {
+	return fmt.Sprintf("Deis controller endpoint %s: expected status code %d, got %d", u.endpoint, u.expected, u.actual)
+}
+
+func controllerURLStr(conf *Config, additionalPath ...string) string {
+	return fmt.Sprintf("http://%s:%s/%s", conf.WorkflowHost, conf.WorkflowPort, strings.Join(additionalPath, "/"))
+}
+
+func setReqHeaders(builderKey string, req *http.Request) {
+	req.Header.Add("Content-Type", contentType)
+	req.Header.Add("Accept", contentType)
+	req.Header.Add("User-Agent", userAgent)
+	req.Header.Add("X-Deis-Builder-Auth", builderKey)
+}
+
+func getAppConfig(conf *Config, builderKey, userName, appName string) (*pkg.Config, error) {
+	url := controllerURLStr(conf, "v2", "hooks", "config")
+	data, err := json.Marshal(&pkg.ConfigHook{
+		ReceiveUser: userName,
+		ReceiveRepo: appName,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	b := bytes.NewReader(data)
+	req, err := http.NewRequest("POST", url, b)
+	if err != nil {
+		return nil, err
+	}
+
+	setReqHeaders(builderKey, req)
+
+	log.Debug("Workflow request POST /v2/hooks/config\n%s", string(data))
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
```

Comment:
Is this working?  Last time we discussed , you faced an issue here and decided to use get request.